python/smqtk/algorithms/descriptor_generator/caffe_descriptor.py

```diff
@@ -408,8 +408,9 @@ def _process_batch(self, uuids4proc, data_elements, descr_elements, procs):
         self._log.debug("extracting return layer '%s' into descriptors",
                         self.return_layer)
         for uid, v in zip(uuids4proc, self.network.blobs[self.return_layer].data):
-            descr_elements[uid].set_vector(v)
-
+            # In case caffe generates multidimensional array (rows, 1, 1)
+            # or matrix format result
+            descr_elements[uid].set_vector(numpy.array(v, copy=False).flatten())
```

Comment:
Turns out I was wrong again. ``flatten`` is going to always make a copy, which might not be the best idea. Might end up being better to check ``v``'s shape first to decide whether flattening is necessary:
```
...
if v.ndim > 1:
    v = numpy.array(v, copy=False).flatten()
descr_elements[uid].set_vector(v)
...
```